bundles/org.openhab.binding.neohub/src/main/java/org/openhab/binding/neohub/internal/NeoHubHandler.java

```diff
@@ -244,10 +288,18 @@ private void fastPollingSchedulerExecute() {
      * 
      */
     public boolean isConfigured(String deviceName) {
-        if (lastInfoResponse == null) {
+        boolean noInfoResponse;
+        synchronized (this) {
+            noInfoResponse = cachedInfoResponse == null;
+        }
+        if (noInfoResponse) {
             fromNeoHubFetchPollingResponse();
         }
-        return lastInfoResponse != null && lastInfoResponse.getDeviceInfo(deviceName) != null;
+        NeoHubInfoResponse myInfoResponse;
+        synchronized (this) {
+            myInfoResponse = cachedInfoResponse;
+        }
+        return myInfoResponse != null && myInfoResponse.getDeviceInfo(deviceName) != null;
```

Comment:
What you should do here instead is assign cachedInfoResponse to a local variable THEN null check the variable. These two synchronized blocks aren't doing any good here.

Reply:
eliminated cachedInfoResponse